<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>829</width>
    <height>999</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QScrollArea" name="scrollArea">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>120</y>
      <width>661</width>
      <height>471</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOn</enum>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents_3">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>645</width>
       <height>455</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="crop_button">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>900</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>crop</string>
    </property>
   </widget>
   <widget class="image_processing" name="original_image">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>590</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="image_processing" name="image_canvas">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>590</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="redo_button">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Redo</string>
    </property>
   </widget>
   <widget class="QPushButton" name="undo_button">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>920</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Undo</string>
    </property>
   </widget>
   <widget class="QDial" name="dial">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>850</y>
      <width>100</width>
      <height>100</height>
     </rect>
    </property>
    <property name="maximum">
     <number>360</number>
    </property>
    <property name="singleStep">
     <number>10</number>
    </property>
    <property name="pageStep">
     <number>20</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QSlider" name="zoom_slider">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>740</y>
      <width>541</width>
      <height>71</height>
     </rect>
    </property>
    <property name="minimum">
     <number>20</number>
    </property>
    <property name="maximum">
     <number>200</number>
    </property>
    <property name="pageStep">
     <number>30</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="save_button">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>20</y>
      <width>161</width>
      <height>91</height>
     </rect>
    </property>
    <property name="text">
     <string>save</string>
    </property>
   </widget>
   <widget class="QPushButton" name="back_button">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>20</y>
      <width>131</width>
      <height>61</height>
     </rect>
    </property>
    <property name="text">
     <string>back</string>
    </property>
   </widget>
   <widget class="QPushButton" name="reset_button">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>920</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>reset</string>
    </property>
   </widget>
   <widget class="image_processing" name="backup_image">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>590</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>image_processing</class>
   <extends>QLabel</extends>
   <header location="global">image_processing.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
